internal/test/framework/framework.go

```diff
@@ -175,3 +177,15 @@ func (f *TestFramework) TearDown() {
 		}
 	}
 }
+
+// WriteToArtifactDir will write contents to $ARTIFACT_DIR/filename
+func (f *TestFramework) WriteToArtifactDir(contents []byte, filename string) error {
+	path := filepath.Join(artifactDir, filename)
+	dir, _ := filepath.Split(path)
+	err := os.MkdirAll(dir, os.ModePerm)
```

Comment:
Do you need to create the artifact directory? I thought it was created for us by the CI framework. Either way creating a directory per VM will take care of this.

Reply:
I needed to create it as sub tests contain "/" which end up becoming sub directories